### PR TITLE
Fix adding the API documentation to the generated resource function

### LINKED ISSUE #53: Add the API documentation to the generated resource function
**Description:**
Add the API documentation for resource function that generate in the service file using details given in the openapi specification. 
Example :
```ballerina
# This is a documentation about the resource (description given in the openapi spec)
#
#    + caller- description 
#    + req - description
#
resource function get pets(http:Caller caller, http:Request req) returns error? {     
}
```

**Suggested Labels:**

**Suggested Assignees:**

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### PR DESCRIPTION
**Pull request by lnash94**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: openapi-cli/src/test/resources/requestBody.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        200:
+          description: OK
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+components:
+  schemas:
+    User:           # <----------
```
Comment (keizer619) [inline openapi-cli/src/test/resources/requestBody.yaml:22]: Can we remove this?

File: openapi-cli/src/main/java/module-info.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
     requires swagger.models;
     requires swagger.parser;
     requires swagger.parser.v2.converter;
+    requires io.swagger.v3.oas.models;
```
Comment (hemikak) [inline openapi-cli/src/main/java/module-info.java:36]: Shall we keep these in alphabetical order ?

File: openapi-cli/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiOperation.java [hunk 3]
```diff
@@ -27,6 +27,17 @@
     private String opName;
     private List<BallerinaOpenApiParameter> parameterList;
     private BallerinaOpenApiRequestBody requestBody;
+    private String operationDescription;
+
+    public String getOperationDescription() {
+
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiOperation.java:33]: Remove empty line

File: openapi-cli/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiOperation.java [hunk 4]
```diff
@@ -27,6 +27,17 @@
...
+        return operationDescription;
+    }
+
+    public void setOperationDescription(String operationDescription) {
+
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiOperation.java:38]: Remove empty line

File: openapi-cli/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiParameter.java [hunk 5]
```diff
@@ -26,6 +26,17 @@
     private String paramName;
     private String refType;
     private BallerinaOpenApiSchema paramType;
+    private String paramDescription;
+
+    public String getParamDescription() {
+
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiParameter.java:32]: Remove empty line

File: openapi-cli/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiParameter.java [hunk 6]
```diff
@@ -26,6 +26,17 @@
...
+        return paramDescription;
+    }
+
+    public void setParamDescription(String paramDescription) {
+
```
Comment (hemikak) [inline openapi-cli/src/main/java/org/ballerinalang/openapi/typemodel/BallerinaOpenApiParameter.java:37]: Remove empty line

File: openapi-cli/src/test/resources/requestBody.yaml (hunk 1 above)
Comment (lnash94) [reply openapi-cli/src/test/resources/requestBody.yaml:22]: sure , we can, I will remove

File: openapi-cli/src/test/resources/expected_gen/generatedRB.bal [hunk 7]
```diff
@@ -0,0 +1,13 @@
+service / on ep0 {
+
+#
+# + caller - Caller client object represents the endpoint
+# + req    - Req represents the message which came along over the network
+
+# + payload - Request body payload.
+# + return - Error value if an error occurred, and return nil otherwise
+    resource function post requestBody(http:Caller caller, http:Request req
+    , @http:Payload {} User  payload) returns error? {}
```
Comment (hemikak) [inline openapi-cli/src/test/resources/expected_gen/generatedRB.bal:10]: Where is the user record ?

Comment (lnash94) [reply openapi-cli/src/test/resources/expected_gen/generatedRB.bal:10]: The user record is generated with the separate schema file.

**Comment by lnash94** [comment]
Windows github workflow failing with know issue,  Please refer to this issue [https://github.com/ballerina-platform/ballerina-openapi/issues/57](https://github.com/ballerina-platform/ballerina-openapi/issues/57)

**Review by hemikak — APPROVED** [review]

**Review by keizer619 — APPROVED** [review]

**Pull request merged**